Formula/g/gtkmm4.rb

```diff
@@ -34,10 +36,37 @@ class Gtkmm4 < Formula
   depends_on "libsigc++"
   depends_on "pangomm"
 
+  uses_from_macos "m4" => :build # Needed to apply the patch.
+  uses_from_macos "perl" => :build # Needed to apply the patch.
+
+  on_linux do
+    depends_on "perl-xml-parser" => :build # Needed to apply the patch.
+  end
+
+  # Fix build failure with Gtk4 >=4.20.0
+  # https://gitlab.gnome.org/GNOME/gtkmm/-/commit/94959145f8b9248e7f6384fb293f1429599f614d
+  # TODO: When removing this patch, also remove the following build deps:
+  #       - mm-common
+  #       - m4
+  #       - perl
+  #       - perl-xml-parser
+  patch do
+    url "https://gitlab.gnome.org/GNOME/gtkmm/-/commit/94959145f8b9248e7f6384fb293f1429599f614d.diff"
+    sha256 "509ebd878b30a84becff4329bfb8096d65a8d88200bf831f9e3fdb33d877921d"
+  end
+
   def install
-    system "meson", "setup", "build", *std_meson_args
+    # We set `maintainer-mode=true` to apply the patch.
+    # We set `build-documentation=false` because it gets enabled by maintainer-mode.
+    # TODO: Remove `args` array and `mm-common`/`m4`/`perl`/etc deps when patch is no longer needed.
+    args = %w[-Dmaintainer-mode=true -Dbuild-documentation=false]
+    odie "Check if `args` array and extra build deps can be removed!" if build.bottle? && version > "4.18.0"
+    system "meson", "setup", "build", *args, *std_meson_args
     system "meson", "compile", "-C", "build", "--verbose"
     system "meson", "install", "-C", "build"
+
+    # TODO: Remove the line below before merging.
+    logs.install "untracked"
```

Comment:
Tried to build this locally so I can see the changes to the generated files, but the source tarball download seems to keep failing.

Adding this temporarily so I can easily get the changes and create a patch.